aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -35,6 +36,75 @@ impl<'a, S: StateView> StorageAdapter<'a, S> {
             .get_state_value(&StateKey::AccessPath(access_path.clone()))
             .map_err(|_| PartialVMError::new(StatusCode::STORAGE_ERROR))
     }
+
+    pub fn get_resource_group(&self, struct_tag: &StructTag) -> Result<Option<StructTag>, VMError> {
+        let module_bytes = if let Some(module_bytes) = self.get_module(&struct_tag.module_id())? {
```

Comment:
This is likely to inefficient for the final solution? Metadata should be really retrieved from the VM which already holds the deserialized module in the loader cache.

Reply:
Possibly, we'd likely want to make this as realistic as possible since this will add some nuances into our gas costs. I have a few other concerns such as how do we make sure we charge the reads from storage appropriately. We don't want to just charge them for the resource they explicitly read, but the entirety of the group.

Reply:
I've resolved this in the following PR: https://github.com/aptos-labs/aptos-core/pull/6040